### ISSUE TITLE
Fail to add new `event_categories` to an existing `aws_db_event_subscription`

### ISSUE THREAD

**Issue by vnourdin** [hashicorp/terraform-provider-aws#21319]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
```
Terraform v0.15.3
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.63.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```
### Affected Resource(s)
* aws_db_event_subscription

### Terraform Configuration Files
I have a Terraform module that do a db event subscription. I try to add new event categories to subscribe to, but it fails.
```hcl
locals {
  rds_event_subscription_name = "${var.rds_name}-rds-event-subscription"
}

resource "aws_db_event_subscription" "rds_event_subscription" {
  name      = local.rds_event_subscription_name
  sns_topic = data.terraform_remote_state.lambda.outputs.sns_topic_arn

  source_ids       = var.event_source_ids
  source_type      = var.event_source_type
  event_categories = var.event_categories
  enabled          = var.enable_event_subscription

  tags = merge(
    var.custom_tags,
    { "name" = local.rds_event_subscription_name }
  )
}
```

### Terraform plan

```terraform
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.my_module.module.rds_event_subscription.aws_db_event_subscription.rds_event_subscription will be updated in-place
  ~ resource "aws_db_event_subscription" "rds_event_subscription" {
      ~ event_categories = [
          + "deletion",
          + "failure",
          + "maintenance",
          + "recovery",
          + "restoration",
            # (1 unchanged element hidden)
        ]
        id               = "my-subscription"
        name             = "my-subscription"
        tags             = {
            "name" = "my-subscription"
        }
        # (7 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Expected Behavior

It should add those events category ad showed by the plan.

### Actual Behavior

```terraform
│ Error: error updating RDS Event Subscription (my-subscription): InvalidParameterCombination: If SourceType is null, Categories must also be null.
│   status code: 400, request id: 44c82231-d1f1-41fe-9423-001ea409a5be
│ 
│   with module.my_module.module.rds_event_subscription.aws_db_event_subscription.rds_event_subscription,
│   on .terraform/modules/my_module.rds_event_subscription/rds-event-subscription/main.tf line 5, in resource "aws_db_event_subscription" "rds_event_subscription":
│    5: resource "aws_db_event_subscription" "rds_event_subscription" {
```

### Steps to Reproduce

1. `terraform apply` a subscription with only one category
2. change your calling code to add more modules
3. `terraform apply`

### References
I see this problem should have been resolved a long time ago :
* #3100

**Comment by vnourdin** [comment]
I just make it works by downgrading to provider `3.40.0`, something happen between those versions.

**Comment by vnourdin** [comment]
Indeed, it's working in 3.60.0 too !
Thanks :pray:

**Comment by kamilturek** [comment]
Fixed by #21338.
